src/main/scala/firrtl/graph/EulerTour.scala

```diff
@@ -0,0 +1,224 @@
+// See LICENSE for license details.
+
+package firrtl.graph
+
+import scala.collection.mutable
+
+/** Euler Tour companion object */
+object EulerTour {
+  /** Create an Euler Tour of a `DiGraph[T]` */
+  def apply[T](diGraph: DiGraph[T], start: T): EulerTour[Seq[T]] = {
+    val r = mutable.Map[Seq[T], Int]()
+    val e = mutable.ArrayBuffer[Seq[T]]()
+    val h = mutable.ArrayBuffer[Int]()
+
+    def tour(u: T, parent: Seq[T] = Seq(), height: Int = 0): Unit = {
+      val id = parent :+ u
```

Comment:
I know that premature optimization is the root of all evil, but `Seq()` returns a `List` which is obviously not the best thing to append to. it would be better if this could either be prepending or just use a Vector

since tour is only called with default arguments once and is nested function, perhaps ditch the default args?

Reply:
Premature optimization of this class is likely a good thing. Any performance bugs here usually get amplified (as you either don't use any WiringTransforms or you use a lot of them, e.g., scan chain insertion). 

Yes, I don't think the defaults are needed here.